core/src/main/scala/akka/persistence/r2dbc/internal/BySliceQuery.scala

```diff
@@ -416,20 +420,30 @@ import org.slf4j.Logger
       state.backtracking && state.rowCount < settings.querySettings.bufferSize - state.backtrackingExpectFiltered
     }
 
+    def switchToBacktracking(state: QueryState, newIdleCount: Long): Boolean = {
+      // Note that when starting the query with offset = NoOffset it will switch to backtracking immediately after
+      // the first normal query because between(latestBacktracking.timestamp, latest.timestamp) > halfBacktrackingWindow
+
+      val qSettings = settings.querySettings
+      val previousTimestamp =
+        if (state.previous == TimestampOffset.Zero) state.latest.timestamp else state.previous.timestamp
+
+      qSettings.backtrackingEnabled &&
+      !state.backtracking &&
+      state.latest != TimestampOffset.Zero &&
+      // no backtracking if far behind current wall clock time
+      previousTimestamp.isAfter(clock.instant().minus(firstBacktrackingQueryWindow)) &&
```

Comment:
I don't think it has to be more exact than this.

I have also verified that this works as expected with a sample app. Generated some 100k events test data. Ran catch up without triggering any backtracking, and then storing some new events that trigger backtracking again, and after a while backtracking stops again.

Reply:
No, there is one problem. Let's say that a projection is stopped/crashed and then isn't started immediately again. Then it should still run backtracking from the start offset (minus some) until the backtracking offset is ahead of the start offset.